app/src/main/java/com/rajatgoyal/puzzle15/ui/GameActivity.java

```diff
@@ -537,8 +537,11 @@ public void onSwipeBottom() {
      * @param colMove number of columns to move
      */
     private void makeMove(int rowMove, int colMove) {
-        int newEmptyRowIndex = emptyRowIndex + rowMove;
-        int newEmptyColIndex = emptyColIndex + colMove;
+        // abs sum ensures that one of them is only one of them is 1 or -1
+        if (Math.abs(rowMove) + Math.abs(colMove) != 1) return;
+
+        int newEmptyRowIndex = gameMatrix.getEmptyCellRow() + rowMove;
+        int newEmptyColIndex = gameMatrix.getEmptyCellCol() + colMove;
```

Comment:
There's no `gameMatrix` available in this file. Use `emptyRowIndex` and `emptyColIndex` only. Change them in the other branch.